### PR TITLE
deprecate udp_listener & tcp_listener

### PR DESCRIPTION
**Pull request by sparrc**
### Required for all PRs:

- [x] CHANGELOG.md updated (we recommend not updating this until the PR has been approved by a maintainer)
- [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)
- [x] README.md updated (if adding a new plugin)

### CODE REVIEW

File: plugins/inputs/socket_listener/README.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+at least 8MB before trying to run large amounts of UDP traffic to your instance.
+8MB is just a recommendation, and can be adjusted higher.
+
+### Linux
+Check the current UDP/IP receive buffer limit & default by typing the following
+commands:
+
+```
+sysctl net.core.rmem_max
+sysctl net.core.rmem_default
```
Comment (phemmer) [inline plugins/inputs/socket_listener/README.md]: Do we want to add a comment the relation with the `read_buffer_size` config param? If `read_buffer_size` is unset, `net.core.rmem_default` will be used. `read_buffer_size` can be set up to `net.core.rmem_max`.

et al. for the BSD/Darwin settings.

Comment (sparrc) [inline plugins/inputs/socket_listener/README.md]: yes, I will make a note of that

**Comment by sparrc** [comment]
@phemmer how's that?

**Comment by phemmer** [comment]
> @phemmer how's that?

👍 

Looks good to me.

**Comment by sebito91** [comment]
Interesting call to deprecate the existing listeners, is this purely to maintain more compact codebase?

**Comment by sparrc** [comment]
@sebito91 yep, more or less. I think it's better for maintainability & stability if everyone is using the same code.

Of course, udp_listener & tcp_listener won't be going away anytime soon, you'll just get a warning message on startup.

**Comment by sebito91** [comment]
Awesome, thanks @sparrc + @phemmer...looks fine by us and happier to be on the maintained input that deprecated.

FYI, a lot of our users are starting to leverage this endpoint for custom collectors...they just don't want to give up their python interpreter for golang :/ TL; DR is that this is becoming a very useful input for our workflow!

**Pull request merged**